core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala

```diff
@@ -388,6 +388,33 @@ class SparkSubmitSuite
     conf.get("spark.ui.enabled") should be ("false")
   }
 
+  test("handles k8s cluster mode") {
+    val clArgs = Seq(
+      "--deploy-mode", "cluster",
+      "--master", "k8s://host:port",
+      "--executor-memory", "5g",
+      "--class", "org.SomeClass",
+      "--driver-memory", "4g",
+      "--conf", "spark.kubernetes.namespace=spark",
+      "--conf", "spark.kubernetes.driver.docker.image=bar",
```

Comment:
Should we also test the arg "--kubernetes-namespace"?

Reply:
`--kubernetes-namespace` has been removed in this PR.

Reply:
Should also remove https://github.com/apache/spark/pull/19717/files#diff-63a5d817d2d45ae24de577f6a1bd80f9R595?

Reply:
Done.